watchdog/main.go

```diff
@@ -50,8 +52,28 @@ func main() {
 		MaxHeaderBytes: 1 << 20, // Max header of 1MB
 	}
 
-	http.HandleFunc("/_/health", makeHealthHandler())
-	http.HandleFunc("/", makeRequestHandler(&config))
+	redCounter := prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "watchdog_invocation_count",
+		Help: "invocations of main endpoint",
+	}, []string{"status"})
+
+	healthCounter := prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "watchdog_health_invocation_count",
+		Help: "invocations of health endpoint",
+	})
+	redErr := prometheus.Register(redCounter)
+	if redErr != nil {
+		log.Println(redErr)
+	}
+	err := prometheus.Register(healthCounter)
+	if err != nil {
+		log.Println(err)
+	}
+
+	promHTTP := prometheus.Handler()
+	http.Handle("/_/metrics", promHTTP)
```

Comment:
The recommend way of exposing the `metrics` endpoint is with `github.com/prometheus/client_golang/prometheus/promhttp`. The handler should be `http.Handle("/metrics", promhttp.Handler())`

Reply:
This was put out for discussion and you didn't raise an issue with it.

#547

We agreed to use a private namespace for the routes of `/_/` so that it could not clash with the function in the container. Same for the health endpoint.

Reply:
I think there was a reason I moved back to this version of the handler. How to the metrics get registered and associated with the first method/approach?